Bencodex/Encoder.cs

```diff
@@ -9,12 +9,23 @@ namespace Bencodex
 {
     internal static class Encoder
     {
+        private const byte _n = 0x6e;
+        private const byte _t = 0x74;
+        private const byte _f = 0x66;
+        private const byte _i = 0x69;
+        private const byte _c = 0x3a;   // `:`
+        private const byte _e = 0x65;
+        private const byte _u = 0x75;
+        private const byte _l = 0x6c;
+        private const byte _d = 0x64;
+
         // TODO: Needs a unit test.
         public static byte[] Encode(IValue value)
         {
             long estimatedLength = EstimateLength(value);
             var buffer = new byte[estimatedLength];
-            Encode(value, buffer, 0L);
+            long offset = 0;
+            Encode(value, buffer, ref offset);
```

Comment:
What is your intent on these changes? I don't know why you instantiate and use the variable as a `ref`.

Reply:
Maybe I am missing something, but this is the main entry point and the `ref` variable must be declared somewhere? The syntax does not allow passing in `0`. 🤔

Reply:
> The syntax does not allow simply passing in 0 as a number.

Would there be a performance hit to using numbers as they are? Rather, it seems that allocating new memory is worse than using static numbers in terms of memory usage, etc.

Reply:
See the rest of the code and the PR comment. Changed method only creates a single `long` (albeit passed as a `ref`). In the original form, `long` as a value type gets copied over and over and also due to the recursive nature of the algorithm, more local `long`s need to be created and `return`ed and so on. Optimized version has pretty much a fixed size memory usage.